meta/exifTypes.go

```diff
@@ -1,19 +1,32 @@
 package meta
 
 import (
+	"math/bits"
 	"strconv"
 )
 
 //go:generate msgp
 
-// FocalLength is a Focal Length expressed in millimeters.
-type FocalLength float32
-
 var (
+	_ExposureModeIndex    = [...]uint8{0, 4, 10, 22}
+	_ExposureProgramIndex = [...]uint8{0, 11, 17, 27, 47, 72, 93, 112, 120, 129, 133}
+	_MeteringModeIndex    = [...]uint8{0, 7, 14, 37, 41, 51, 64, 71}
```

Comment:
Maybe add these as consts's too?

Reply:
Yes. I did some benchmarking and this method is used by [https://pkg.go.dev/golang.org/x/tools/cmd/stringer](https://pkg.go.dev/golang.org/x/tools/cmd/stringer) and is significantly more performant than using a map. 

Thanks for the updates.

Reply:
My suggestion was regarding the magic value `0, 4, ...` and to on... I would like to replace those by:

```go
const (
	exposureModeUnknown = 0
	exposureModeNormal  = 4
)

var (
	_ExposureModeIndex    = [...]uint8{
		exposureModeUnknown,
		exposureModeNormal,
		...,
	}
)
```
It's much more readable, have no performance-impact and less prone to errors - but it's really not a big deal.

Reply:
I appreciate your suggestions and have made the changes to ExposureMode, ProgramMode, and MeteringMode.

Reply:
The "performance" of looking up a string from 2 slices is faster than a map. this doesn't make a difference unless you scale it to very large numbers. An example of this is serving thousands of exif responses over json.